packages/actionmenu/src/js/index.js

```diff
@@ -1,62 +1,34 @@
 import * as vars from '../vars/index.js'
+import { layout } from '@pluralsight/ps-design-system-core'
 
-export function calcNextIndex(childrenProps, direction, currentIndex) {
-  const count = childrenProps.length
-  const next = currentIndex + direction
-
-  if (next < 0) return 0
-  if (next > count - 1) return count - 1
-
-  const isDisabled = childrenProps[next].disabled
-  if (isDisabled) {
-    return direction === 1
-      ? find(
-          i => ++i,
-          i => i >= childrenProps.length - 1,
-          childrenProps,
-          currentIndex,
-          currentIndex
-        )
-      : find(
-          i => --i,
-          i => i < 0,
-          childrenProps,
-          currentIndex,
-          currentIndex
-        )
-  } else {
-    return next
-  }
-}
-
-function find(iterate, base, props, start, from) {
-  const i = iterate(from)
-  if (base(i)) {
-    return start
-  } else if (props[i].disabled) {
-    return find(iterate, base, props, start, i)
-  } else {
-    return i
-  }
-}
-
-export function calcNestedMenuPosition(menuWidth, origin) {
+export function calcNestedMenuPosition(parent, menu, origin) {
   return {
     topLeft: {
-      left: menuWidth + vars.style.nestedMenuHorzOverlap,
+      left: -parent.width,
       top: `calc(-1 * ${vars.style.menuPaddingVert})`
     },
     topRight: {
-      right: menuWidth + vars.style.nestedMenuHorzOverlap,
+      right: parent.width,
       top: `calc(-1 * ${vars.style.menuPaddingVert})`
     },
     bottomRight: {
-      right: menuWidth + vars.style.nestedMenuHorzOverlap,
-      bottom: `calc(-1 * ${vars.style.menuPaddingVert})`
+      right: parent.width,
+      top: -Math.abs(parent.height - menu.height)
     },
     bottomLeft: {
-      left: menuWidth + vars.style.nestedMenuHorzOverlap,
-      bottom: `calc(-1 * ${vars.style.menuPaddingVert})`
+      left: -parent.width,
+      top: -Math.abs(parent.height - menu.height)
     }
   }[origin]
 }
+
+export const ellipsesStyles = {
+  whiteSpace: 'nowrap',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis'
+}
+
+export const iconLeftStyles = {
+  marginRight: layout.spacingXSmall,
+  flex: 'none'
+}
```

Comment:
Not sure about these there styles that make adding things other than default text easy to style and align. Was considering adding a `iconRightStyles`  but wanted feedback on this approach first.